rust-runtime/aws-smithy-types/src/base64.rs

```diff
@@ -151,9 +151,17 @@ fn decode_inner(inp: &str) -> Result<Vec<u8>, DecodeError> {
     Ok(ret)
 }
 
+/// Given the length of some data in bytes, return how many bytes it would take to base64 encode
+/// that data.
+pub fn encoded_length(length: u64) -> u64 {
+    (length + 2) / 3 * 4
```

Comment:
I'm unsure about this. Was the `+ 2` intended to round up to a multiple of four? If so, I don't think it will accomplish that. Can you modify `test_base64_encoded_length` to be a `proptest` to show that this works more broadly?

Reply:
I don't remember where this impl came from but it does work. Adding a proptest is a good callout though, so I added one:
```rust
        #[test]
        fn encoded_length_is_correct(v in any::<Vec<u8>>()) {
            let correct = encode(v.as_slice()).len() as u64;
            let ours = encoded_length(v.len() as u64);
            assert_eq!(ours, correct);
        }
```